packages/wasm-ast-types/src/react-query.ts

```diff
@@ -115,28 +130,53 @@ export const createReactQueryHook = ({
                                     t.stringLiteral(hookKeyName),
-                                    t.memberExpression(
+                                    t.optionalMemberExpression(
                                         t.identifier('client'),
-                                        t.identifier('contractAddress')
-                                    )
+                                        t.identifier('contractAddress'),
+                                        false,
+                                        !!options?.optionalClient
+                                    ),
                                 ]),
                                 t.arrowFunctionExpression(
                                     [],
-                                    t.callExpression(
-                                        t.memberExpression(
-                                            t.identifier('client'),
-                                            t.identifier(methodName)
+                                    optionalConditionalExpression(
+                                        t.identifier('client'),
+                                        t.callExpression(
+                                            t.memberExpression(
+                                                t.identifier('client'),
+                                                t.identifier(methodName)
+                                            ),
+                                            args
                                         ),
-                                        args
+                                        t.identifier('undefined'),
+                                        !!options?.optionalClient
                                     ),
                                     false
                                 ),
-                                t.identifier('options')
+                                t.objectExpression([
+                                    t.objectProperty(
+                                        t.identifier('enabled'),
+                                        t.logicalExpression(
+                                            '&&',
+                                            t.unaryExpression(
+                                                '!',
+                                                t.unaryExpression('!', t.identifier('client'))
+                                            ),
+                                            t.optionalMemberExpression(
+                                                t.identifier('options'),
+                                                t.identifier('enabled'),
+                                                false,
+                                                true
+                                            )
+                                        )),
+                                    t.spreadElement(t.identifier('options'))
+                                ]),
```

Comment:
Can clean this up if desired!

Reply:
Right now, it generates: `enabled: !!client && options?.enabled,` even if the client is definitely there.

Reply:
AST GANG <3